From 687dbdb771bbe28509f8afd934b3afe41b887e8f Mon Sep 17 00:00:00 2001
From: Huacai Chen <chenhc@lemote.com>
Date: Wed, 1 Aug 2018 14:39:20 +0800
Subject: [PATCH 080/131] drm/radeon: Modify GART TLB setting to fix kdump
 failure

After commit 0986c1a55ca64b44ee12 ("drm/radeon: stop poisoning the GART
TLB") kdump fails to boot the second kernel, so partially revert to the
old behavior.

Cc: stable@vger.kernel.org
Signed-off-by: Huacai Chen <chenhc@lemote.com>
---
 drivers/gpu/drm/radeon/rs600.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/radeon/rs600.c b/drivers/gpu/drm/radeon/rs600.c
index 9fbbe95266fc..3c67aaf44520 100644
--- a/drivers/gpu/drm/radeon/rs600.c
+++ b/drivers/gpu/drm/radeon/rs600.c
@@ -648,6 +648,9 @@ uint64_t rs600_gart_get_page_entry(uint64_t addr, uint32_t flags)
 		addr |= R600_PTE_WRITEABLE;
 	if (flags & RADEON_GART_PAGE_SNOOP)
 		addr |= R600_PTE_SNOOPED;
+#ifdef CONFIG_CRASH_DUMP
+	addr |= R600_PTE_VALID | R600_PTE_READABLE | R600_PTE_WRITEABLE;
+#endif
 	return addr;
 }
 
-- 
2.31.1

